src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -74,12 +74,16 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        Person personToEdit = lastShownList
+                .stream()
+                .filter(person -> person.getId().value == (this.id.value))
+                .findFirst()
+                .orElse(null);
+        if (personToEdit == null) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_ID);
         }
-
-        Person personToEdit = lastShownList.get(index.getZeroBased());
```

Comment:
I like the way you rewrote the code for personToEdit and handled any null values with exceptions.